runtime/lua/vim/filetype.lua

```diff
@@ -2329,6 +2325,18 @@ local pattern = {
     ['^Neomuttrc'] = detect_neomuttrc,
     ['%.neomuttdebug'] = 'neomuttlog',
   },
+  ['xkb'] = {
+    ['/xkb/compat/'] = detect_xkb,
+    ['/xkb/geometry/'] = detect_xkb,
+    ['/xkb/keycodes/'] = detect_xkb,
+    ['/xkb/symbols/'] = detect_xkb,
+    ['/xkb/types/'] = detect_xkb,
+    ['/.xkb/compat/'] = detect_xkb,
+    ['/.xkb/geometry/'] = detect_xkb,
+    ['/.xkb/keycodes/'] = detect_xkb,
+    ['/.xkb/symbols/'] = detect_xkb,
+    ['/.xkb/types/'] = detect_xkb,
```

Comment:
Missing escaping for the `.`
```suggestion
  ['/%.?xkb/'] = {
    ['/%.?xkb/compat/'] = detect_xkb,
    ['/%.?xkb/geometry/'] = detect_xkb,
    ['/%.?xkb/keycodes/'] = detect_xkb,
    ['/%.?xkb/symbols/'] = detect_xkb,
    ['/%.?xkb/types/'] = detect_xkb,
```